### ISSUE TITLE
DOC: Surprising behavior on DataFrame.squeeze() with 0 or 1 col/row

### ISSUE THREAD

**Issue by TomAugspurger** [pydata/pandas#8999]
```python
pd.DataFrame({"A": [1]}).squeeze()  # -> 1
pd.DataFrame({"A": []}).squeeze()  # Noop
```

The second one is genuinely surprising; I would have expected it to return an empty Series with the name set to `A`. But it is consisent with numpy (`np.array([[], []]).squeeze()`) so I've marked is as a doc issue.

**Comment by Winterflower** [comment]
@TomAugspurger just testing this issue with pandas version 0.16.2 and I get an empty DataFrame when I do 

```python
df = pd.DataFrame({'A':[]})
print df.squeeze()

>>>
Empty DataFrame
Columns: [A]
Index: []
```

Is this what you meant by Noop?

**Comment by TomAugspurger** [comment]
Sorry yeah, I should have been clearer. I meant the output was the same as the input (an Empty DataFrame).

**Comment by Winterflower** [comment]
Ah ok, you mean Noop as in no-operation?

**Comment by TomAugspurger** [comment]
Yep

**Comment by jreback** [comment]
closed by #11230
